src/include/migraphx/op/reduce_op.hpp

```diff
@@ -80,11 +80,12 @@ template <class Derived>
 struct reduce_op : op_name<Derived>
 {
     std::vector<std::int64_t> axes{};
+    bool noop_with_empty_axes = false;
```

Comment:
We want to make reduce_op to always be a noop with empty axes. For this PR, just skip the handling of empty axes for now. In a future PR, we can update the onnx parser to insert extra operators for empy axes that is not a noop.

Reply:
Do you want the empty axes handling to be skipped, or changed even in the case when axes are constant or an attribute?
In the current code, that is handled according to ONNX spec.

Reply:
This field should be removed. And we should by default be a noop wit empty axes and the parser can insert extra operators to handle the empty case for the ONNX spec.

Reply:
In the constant axes case the parser already handles that, by not inserting an operator at all, and just passing the input back as the output.
In the variable axes case it cannot be known until runtime whether a noop will occur until the runtime value of axes is checked. 
So which operators could the parser insert in the variable axes case to handle it. The If operator?

Reply:
This field needs to be removed from the operator.

> In the variable axes case it cannot be known until runtime whether a noop will occur until the runtime value of axes is checked.  So which operators could the parser insert in the variable axes case to handle it. The If operator?

We can insert operators such as `equal` to detect this and then insert a `where` operator to pick between the reduce and noop.

Reply:
The Where would only have to be inserted when noop_with_empty_axes is true, and be predicated on the size of the variable axes input. For empty axes the input tensor would need to be returned, while for non-empty axes a reduce would be performed. 
This leads to an issue with Where due to the difference of output dimensions of the two branches, given that one would have the shape of the input, and the other would have whichever shape the variable input axes dictate.

Reply:
> The Where would only have to be inserted when noop_with_empty_axes is true

`reduce_op` should be a noop with empty axes, by default. So then when `noop_with_empty_axes=false` the `where` would take two different reduces: `is_empty(axes) ? reduce_op(x, computed_axes) : reduce_op(x, axes)`. For both branches the axes will be passed as a variable input(even though the empty case will be taking a literal) so they produce the same output shape.

Reply:
Has this issue been resolved yet?

Reply:
Hi @CharlieL7, not yet, I'd been on vacation for most of December so the PR was left unattended.